static/js/message_scroll.js

```diff
@@ -70,9 +70,10 @@ exports.show_end_of_results_notice = function () {
     // It's a bit hacky to use the href, but
     // !filter.includes_full_stream_history() implies streams:public
     // wasn't already present.
-    $(".all-messages-search-caution a.search-shared-history").attr(
-        "href", window.location.hash.replace("#narrow/", "#narrow/streams/public/")
-    );
+    const operators = narrow_state.filter().operators();
+    const public_operator = {operator: 'streams', operand: 'public'};
+    const update_hash = hash_util.operators_to_hash([public_operator].concat(operators));
+    $(".all-messages-search-caution a.search-shared-history").attr("href", update_hash);
```

Comment:
lgtm; but maybe we should extract this computation as a function and add some unit tests?